app/scripts/metamask-controller.js

```diff
@@ -3494,41 +3494,29 @@ export default class MetamaskController extends EventEmitter {
         this._convertMnemonicToWordlistIndices(seedPhraseAsBuffer),
       );
 
-      // Scan accounts until we find an empty one
-      const { chainId } = this.networkController.state.providerConfig;
+      // Scan accounts and apply checks
       const ethQuery = new EthQuery(this.provider);
       const accounts = await this.keyringController.getAccounts();
-      let address = accounts[accounts.length - 1];
-
-      for (let count = accounts.length; ; count++) {
-        const balance = await this.getBalance(address, ethQuery);
+      let emptyAccountsCounter = 0;
 
-        if (balance === '0x0') {
-          // This account has no balance, so check for tokens
-          await this.detectTokensController.detectNewTokens({
-            selectedAddress: address,
-          });
-
-          const tokens =
-            this.tokensController.state.allTokens?.[chainId]?.[address];
-          const detectedTokens =
-            this.tokensController.state.allDetectedTokens?.[chainId]?.[address];
+      for (
+        let index = accounts.length - 1;
+        emptyAccountsCounter < 10;
+        index++
+      ) {
+        let address = accounts[index];
```

Comment:
To do `accounts[index]`, it would need to re-fetch `accounts = await this.keyringController.getAccounts()` on each iteration to populate the new account.  Otherwise `accounts` only contains the initial account, and `index` goes out of bounds.  But since only 1 account is processed at a time, I think it's easier to just capture the return value from `addNewAccount` into `address`.